app/main.py

```diff
@@ -1,3 +1,77 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.shops import Shop
+from app.cars import Car
+from app.customers import Customer
+
+
+def shop_trip() -> None:
+    list_of_customers = []
+    list_of_shops = []
+
+    with open("app/config.json", "r") as file_info:
+        all_info = json.load(file_info)
+
+    fuel_price = all_info["FUEL_PRICE"]
+
+    for customer in all_info["customers"]:
+        customer_car = Car(customer["car"]["brand"],
+                           customer["car"]["fuel_consumption"])
+
+        list_of_customers.append(
+            Customer(
+                customer["name"],
+                customer["product_cart"],
+                customer["location"],
+                customer["money"],
+                customer_car
+            )
+        )
+
+    for shop in all_info["shops"]:
+        list_of_shops.append(
+            Shop(
+                shop["name"],
+                shop["location"],
+                shop["products"]
+            )
+        )
+
+    costs_for_trips = {}
+
+    for customer in list_of_customers:
+        print(f"{customer.name} has {customer.money} dollars")
+        for shop in list_of_shops:
+            cost_for_trip = round(
+                2 * customer.car.calculate_trip_cost(
+                    customer.car.calculate_distance(
+                        customer.location,
+                        shop.location
+                    ),
+                    fuel_price
+                )
+                + shop.calculate_cost_for_products(customer.product_cart)
+                , 2
+            )
+            print(f"{customer.name}'s trip to the {shop.name} costs "
+                  f"{cost_for_trip}")
+            costs_for_trips[shop] = cost_for_trip
+
+        if min(costs_for_trips.values()) > customer.money:
+            print(f"{customer.name} "
+                  f"doesn't have enough money to make a purchase in any shop")
+        else:
+            chippest_shop = list(costs_for_trips.keys())[
```

Comment:
Please find another, simple way to find `chippest_shop`
Maybe you can do it during iteration over `list_of_shops`